core-bundle/contao/classes/Crawl.php

```diff
@@ -360,6 +363,42 @@ private function generateSubscribersWidget(array $subscriberNames): Widget
 		return $widget;
 	}
 
+	private function generateMaxDepthWidget(): Widget
+	{
+		$name = 'crawl_depth';
+
+		$widget = new SelectMenu();
+		$widget->id = $name;
+		$widget->name = $name;
+		$widget->label = $GLOBALS['TL_LANG']['tl_maintenance']['crawlDepth'][0];
+		$widget->setInputCallback($this->getInputCallback($name));
+
+		$options = array();
+		$default = System::getContainer()->getParameter('contao.crawl.max_depth');
+
+		for ($i = 1; $i <= 20; ++$i) {
```

Comment:
@Toflar I deliberately did not include `0` here, so a user cannot set the depth to "unlimited". But maybe we want to allow this?

Reply:
imho anything lower than 3 also does not make sense, otherwise nothing will be really crawled:

* Level 1: https://example.com/robots.txt
* Level 2: https://example.com/sitemap.xml
* Level 3: https://example.com/<url-from-sitemap>

Reply:
I agree that lower than 3 makes no sense and 20 is ridiculously high. I would never go deeper than 6 personally. We can still increase it if anybody comes up with an acceptable use case. As @fritzmg has already shown, all the URLs that are correctly part of the sitemap would be on level 3 and I would always argue and say "okay, your page is so important that you want to crawl until a depth of 7 - why is it not in your sitemap then if it's so important?"

Reply:
Changed in f90a9d1dda2f30db563a32e05769786530c7d475.